onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -446,16 +454,22 @@ Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, cudnnRe
                       tmp_div, tmp_div,
                       input_data, input_count);
     } else if (log_sum_exp_) {
-      // Reduce max -- Max/Min will output indices data
-      CudnnReduceDescriptor reduce_max_desc;
-      ORT_RETURN_IF_ERROR(reduce_max_desc.Set(CUDNN_REDUCE_TENSOR_MAX, cudnn_type_X, CUDNN_REDUCE_TENSOR_NO_INDICES));
-      size_t indices_bytes_max = 0;
-      CUDNN_RETURN_IF_ERROR(cudnnGetReductionIndicesSize(CudnnHandle(), reduce_max_desc, input_tensor, output_tensor, &indices_bytes_max));
-      auto indices_cuda_max = GetScratchBuffer<uint32_t>(indices_bytes);
-      CUDNN_RETURN_IF_ERROR(cudnnReduceTensor(
-          CudnnHandle(), reduce_max_desc, indices_cuda_max.get(), indices_bytes_max, workspace_cuda.get(), workspace_bytes,
-          &one, input_tensor, reinterpret_cast<const CudaT*>(X->template Data<T>()),
-          &zero, output_tensor, reinterpret_cast<CudaT*>(Y->template MutableData<T>())));
+      if (input_count == output_count) {
+        if (Y->template MutableData<T>() != X->template Data<T>()) {
+          CUDA_RETURN_IF_ERROR(cudaMemcpyAsync(Y->template MutableData<T>(), X->template Data<T>(), input_count * sizeof(T), cudaMemcpyDeviceToDevice));
+        }
```

Comment:
Would be helpful to add a comment saying why it's valid to just copy the input if there's only 1 value

Reply:
Also good to explain when input_count == output_count. Scalar or batch of scalars right?

Reply:
With reduction, it is usually when the only one element remaining to reduce and the output would be that same element. So the same count for input and output produces an error condition for cudnnReduceTensor

---
In reply to: [394739430](https://github.com/microsoft/onnxruntime/pull/3260#discussion_r394739430) [](ancestors = 394739430)